### DISCUSSION TITLE
Avoiding "Invalid Date" message with {{ date }} when field is empty

### DISCUSSION THREAD

**Discussion by sergey-dme** [Budibase/budibase#1971] [Help] [answered]
Hi all, Is there a way to avoid "Invalid Date" message when Date/Time filed is empty and` {{ date }} `used to format it (in Repeater->Headline)?

Thank you.

p.s. Sorry for a number of discussions opened today.

**Comment by mike12345567** [comment]
Hi @sergey-dme using `{{ date }}` simply tries to make use of the `date` Handlebars helper however you need to provide some data for it format. For example you can use `{{ date now "DD/MM/YYYY" }}` to format a date like 11/07/2021.

If you have a date within an internal/external table then you might use something like `{{ date New Repeater.table.dateField "DD/MM/YYYY" }}`

For more information about all the different options available for date formatting with the date helper you can look here: https://day.js.org/docs/en/display/format - under the hood we use dayjs to format our dates.

**Comment by sergey-dme** [comment, marked as answer]
Hi @mike12345567  Thanks. I meant, for example there is a filed ECD which is Date/Time type and if I use `{{ date Repeater.Table.ECD "DD/MM/YYYY" }}` to format for my printing-form in Headline (inside Repeater) it returns "Invalid Date" if the filed is empty in the table. I think this is wrong, it should returns nothing if the filed is empty

**Reply by mike12345567** [reply]
Ah I see, if the field is empty this might be the behaviour right now, this might require a fix so that if no date is provided it just doesn't format it, but for now you could do something like `{{#if Repeater.Table.ECD }}{{ date Repeater.Table.ECD "DD/MM/YYYY" }}{{/if}}` - only conditionally formatting the date if it exists.

**Reply by sergey-dme** [reply]
@mike12345567  Cool! Thanks a lot! That is what I asked for :)
Just realised that you just discovered powerful `#if` for me

**Reply by mike12345567** [reply]
Yeap theres some real power within the Handlebars syntax, especially for building up strings! I recommend having a look at the language guide [here](https://handlebarsjs.com/guide/) if you have the time sometime, it can be very enlightening! 🙂
